bbl_screen-patch/patches/printerui/qml/settings/HardwarePage.qml

```diff
@@ -366,9 +349,18 @@ Rectangle {
             anchors.left: parent.left
             anchors.right: parent.right
             columns: 3
-            
             property var imageHeight: 25
-
+            function resetComboBoxDefaults() {
+                var defaults = X1Plus.GpioKeys.resetToDefaultActions(); 
+                
+                for (var i = 0; i < buttonGrid.children.length; i++) {
+                    var loader = buttonGrid.children[i];
+                    if (loader.item) { 
+                        loader.item.defaultSelection = X1Plus.GpioKeys.getDefault(loader.item.btn, loader.item.pressType);
+                        loader.item.currentIndex = loader.item.defaultSelection;
+                    }
+                }
+            }
```

Comment:
hm, I would rather that `GpioKeys` had the definitive model of what's going on, and this be just a view onto that model.  so in that world we would just `X1Plus.GpioKeys.resetToDefaultActions()`, and then the `item`s would happily just reset themselves based on a binding?

Reply:
That's a good suggestion. There's are several sections of code throughout this branch that are very much "let's just get this working and be done" themed and this is one of them.